DirectProgramming/DPC++FPGA/Tutorials/Tools/board_test/README.md

```diff
@@ -0,0 +1,346 @@
+********************* Kernel Launch Test ************************
+*****************************************************************
+
+Launching kernel KernelSender ...
+Launching kernel KernelReceiver ...
+  ... Waiting for sender
+Sender sent the token to receiver
+  ... Waiting for receiver
+
+KERNEL_LAUNCH_TEST PASSED
+
+*****************************************************************
+********************  Kernel Latency  **************************
+*****************************************************************
+
+Processed 10000 kernels in 189.0976 ms
+Single kernel round trip time = 18.9098 us
+Throughput = 52.8827 kernels/ms
+Kernel execution is complete
+
+*****************************************************************
+*************  Kernel-to-Memory Read Write Test  ***************
+*****************************************************************
+
+Maximum device global memory allocation size is 34359737344 bytes 
+Finished host memory allocation for input and output data
+Creating device buffer
+Finished writing to device buffers 
+Launching kernel MemReadWriteStream ... 
+Launching kernel with global offset : 0
+Launching kernel with global offset : 1073741824
+Launching kernel with global offset : 2147483648
+Launching kernel with global offset : 3221225472
+Launching kernel with global offset : 4294967296
+Launching kernel with global offset : 5368709120
+Launching kernel with global offset : 6442450944
+Launching kernel with global offset : 7516192768
+... kernel finished execution. 
+Finished Verification
+KERNEL TO MEMORY READ WRITE TEST PASSED 
+
+*****************************************************************
+*****************  Kernel-to-Memory Bandwidth  *****************
+*****************************************************************
+
+Note: This test assumes that design was compiled with -Xsno-interleaving option
+
+Size of memory bank 1 = 0x200000000 bytes 
+Size of memory bank 2 = 0x200000000 bytes 
+Size of memory bank 3 = 0x200000000 bytes 
+Size of memory bank 4 = 0x200000000 bytes 
+
+Performing kernel transfers of 4096 MBs on the default global memory (address starting at 0)
+Launching kernel MemWriteStream ... 
+Launching kernel MemReadStream ... 
+Launching kernel MemReadWriteStream ... 
+
+Summarizing bandwidth in MB/s/bank for banks 1 to 4
+ 15862.1  15874.6  15871.7  15879.5  MemWriteStream
+ 17339.5  17339.9  17339.5  17339.8  MemReadStream
+ 16014.4  16018.5  16018.2  16018.5  MemReadWriteStream
+
+Name of the global memory type      :    DDR
+Number Of Interfaces            :    4
+Max Bandwidth (all memory interfaces)   :    76800 MB/s 
+Max Bandwidth of 1 memory interface in board_spec.xml :    19200 MB/s 
+
+It is assumed that all memory interfaces have equal widths. 
+
+BOARD BANDWIDTH UTILIZATION = 85.4672
+Warning : Board bandwidth utilization is less than 90% 
+
+KERNEL-TO-MEMORY BANDWIDTH = 16409.7 MB/s/bank
+
+BOARD TEST PASSED
+ 
+```

```

Comment:
We usually have an additional section for Reference Designs, e.g.:
```
## Additional Design Information
### Source Code Explanation

| File                         | Description
---                            |---
| `gzip.cpp`                   | Contains the `main()` function and the top-level interfaces to the SYCL* GZIP functions.
| `gzip_ll.cpp`                | Low latency variant of the top level file.
| `gzipkernel.cpp`             | Contains the SYCL* kernels used to implement GZIP.
| `gzipkernel_ll.cpp`          | Low-latency variant of kernels.
...

### Compiler Flags Used

| Flag | Description
---    |---
`-Xshardware` | Target FPGA hardware (as opposed to FPGA emulator)
`-Xsparallel=2` | Uses two cores when compiling the bitstream through Quartus®
`-Xsseed=<seed_num>` | Uses a particular seed while running Quartus®, selected to yield the best Fmax for this desgin
`-Xsnum-reorder=6` | On Intel Stratix® 10 SX only, specify a wider data path for read data from global memory
`-Xsopt-arg="-nocaching"` | Specifies that cached LSUs should not be used.
`-DNUM_ENGINES=<1|2>` | Specifies that 1 GZIP engine should be compiled when targeting Intel Arria® 10 GX and two engines when targeting Intel Stratix® 10 SX
```

Reply:
Added section with source file and compiler flag information